packages/server/index.js

```diff
@@ -22,4 +22,17 @@ require && require.extensions && delete require.extensions['.coffee.md']
 // https://github.com/electron/electron/blob/master/docs/api/process.md#processenablepromiseapis
 process.enablePromiseAPIs = process.env.CYPRESS_ENV !== 'production'
 
+// don't emit the NODE_TLS_REJECT_UNAUTHORIZED warning while
+// we work on proper SSL verification
+// https://github.com/cypress-io/cypress/issues/5248
+const originalEmitWarning = process.emitWarning
+
+process.emitWarning = (warning, options) => {
+  if (warning && warning.includes && warning.includes('NODE_TLS_REJECT_UNAUTHORIZED')) {
+    return
+  }
+
+  return originalEmitWarning.call(process, warning, options)
```

Comment:
rather than accepting all arguments manually (such as `options`) which can change based on call signature it's much safer to use `apply` here - since we only care about `warning` parameter for our own purposes.

`return originalEmitWarning.apply(process, arguments)`

Also where are the tests? You could likely spawn a child process, and pull off `stderr` to verify that it's not present in the output.

Reply:
Added tests and moved this code to its own module to clean it up in.